app/cni/cmd/kuma-cni/main.go

```diff
@@ -0,0 +1,266 @@
+package main
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"net"
+	"time"
+
+	"github.com/containernetworking/cni/pkg/skel"
+	"github.com/containernetworking/cni/pkg/types"
+	"github.com/containernetworking/cni/pkg/types/current"
+	"github.com/containernetworking/cni/pkg/version"
+
+	"github.com/kumahq/kuma/pkg/core"
+)
+
+const (
+	podRetrievalMaxRetries = 30
+	podRetrievalInterval   = 1 * time.Second
+)
+
+var (
+	log = core.NewLoggerWithRotation(2, "/tmp/kuma-cni.log", 100, 0, 0).WithName("kuma-cni")
+)
+
+// Kubernetes a K8s specific struct to hold config
+type Kubernetes struct {
+	Kubeconfig        string   `json:"kubeconfig"`
+	ExcludeNamespaces []string `json:"exclude_namespaces"`
+	CniBinDir         string   `json:"cni_bin_dir"`
+}
+
+type PluginConf struct {
+	types.NetConf // You may wish to not nest this type
```

Comment:
Do we wish to not nest it?

Reply:
removed that comment, it's been like this for a couple of years in the old CNI